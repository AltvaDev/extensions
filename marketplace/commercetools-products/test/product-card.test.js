'use strict';

const React = require('react');
const renderer = require('react-test-renderer');

const { mockComponent } = require('./helpers.js');

jest.mock('@contentful/forma-36-react-components', () => ({
  Card: mockComponent('card'),
  EntryCard: mockComponent('entry-card'),
  DropdownList: mockComponent('dropdown-list'),
  DropdownListItem: mockComponent('dropdown-list-item'),
  IconButton: mockComponent('icon-button'),
  CardDragHandle: mockComponent('card-drag-handle')
}));

jest.mock('react-sortable-hoc', () => ({
  SortableHandle: component => component,
  SortableElement: component => component
}));

jest.mock('@apollo/react-components', () => ({
  Query: mockComponent('query')
}));

const { SortableProductCard, ProductCardRenderer } = require('./../src/product-card.js');

describe('SortableProductCard', function() {
  test('renders correctly', function() {
    const tree = renderer
      .create(
        <SortableProductCard
          sku="ABCD1234"
          itemIndex={0}
          removeItem={() => {}}
          locale="en"
          sortable={true}
          projectKey="a-cool-project"
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('ProductCardRenderer', function() {
  test('renders error state', function() {
    const tree = renderer
      .create(
        <ProductCardRenderer
          error={new Error('broken!')}
          sku="ABCD1234"
          index={0}
          removeItem={() => {}}
          locale="en"
          projectKey="a-cool-project"
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders loading state', function() {
    const tree = renderer
      .create(
        <ProductCardRenderer
          loading={true}
          sku="ABCD1234"
          index={0}
          removeItem={() => {}}
          locale="en"
          sortable={true}
          projectKey="a-cool-project"
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders missing product state', function() {
    const data = { products: { results: [] } };

    const tree = renderer
      .create(
        <ProductCardRenderer
          sku="ABCD1234"
          data={data}
          index={0}
          removeItem={() => {}}
          locale="en"
          sortable={true}
          projectKey="a-cool-project"
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('renders with data', function() {
    const data = {
      products: {
        results: [
          {
            id: '2c7f2de9-d30a-42cd-bbca-5783381efb9c',
            masterData: {
              current: {
                categories: [
                  {
                    nameAllLocales: [
                      { locale: 'de', value: 'T-shirts' },
                      { locale: 'en', value: 'T-shirts' },
                      { locale: 'it', value: 'T-shirt' }
                    ]
                  },
                  {
                    nameAllLocales: [
                      { locale: 'de', value: 'Bekleidung' },
                      { locale: 'en', value: 'Clothing' },
                      { locale: 'it', value: 'Abbigliamento' }
                    ]
                  }
                ],
                masterVariant: {
                  sku: 'M0E20000000DOSQ',
                  images: [
                    {
                      url:
                        'https://s3-eu-west-1.amazonaws.com/commercetools-maximilian/products/073315_1_large.jpg'
                    }
                  ]
                },
                nameAllLocales: [
                  { locale: 'de', value: 'Bluse Red Valentino rot' },
                  { locale: 'en', value: 'Shirt Red Valentino red' }
                ]
              }
            }
          }
        ]
      }
    };

    test.each([
      ['', { sortable: false, disabled: false }],
      ['and disabled', { sortable: false, disabled: true }],
      ['and sortable', { sortable: true, disabled: false }],
      ['and sortable and disabled', { sortable: true, disabled: true }]
    ])('%s', function(_, { sortable, disabled }) {
      const tree = renderer
        .create(
          <ProductCardRenderer
            sku="ABCD1234"
            data={data}
            index={0}
            removeItem={() => {}}
            locale="en"
            sortable={sortable}
            disabled={disabled}
            projectKey="a-cool-project"
          />
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
