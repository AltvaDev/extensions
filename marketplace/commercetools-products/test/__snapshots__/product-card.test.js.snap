// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCardRenderer renders error state 1`] = `
<div
  className="card"
>
  <div
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <h1
      style={
        Object {
          "flex": "1 1 auto",
          "fontSize": ".875rem",
          "lineHeight": 1.5,
          "margin": 0,
        }
      }
    >
      Error fetching product. This might be authentication issue or an issue with the commerce tools API.
    </h1>
  </div>
</div>
`;

exports[`ProductCardRenderer renders loading state 1`] = `
<div
  className="entry-card"
  dropdownListElements={
    <React.Fragment>
      <Unknown>
        <Unknown
          onClick={[Function]}
        >
          Remove
        </Unknown>
      </Unknown>
    </React.Fragment>
  }
  loading={true}
  title="ABCD1234"
/>
`;

exports[`ProductCardRenderer renders missing product state 1`] = `
<div
  className="card"
>
  <div
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <h1
      style={
        Object {
          "flex": "1 1 auto",
          "fontSize": ".875rem",
          "lineHeight": 1.5,
          "margin": 0,
        }
      }
    >
      Product missing or inaccessible
    </h1>
    <div
      buttonType="muted"
      className="icon-button"
      iconProps={
        Object {
          "icon": "Close",
        }
      }
      label="Remove reference to product"
      onClick={[Function]}
    />
  </div>
</div>
`;

exports[`ProductCardRenderer renders with data  1`] = `
<div
  cardDragHandleComponent={null}
  className="entry-card"
  contentType="T-shirts"
  dropdownListElements={
    <React.Fragment>
      <Unknown>
        <Unknown
          isDisabled={false}
          onClick={[Function]}
        >
          Remove
        </Unknown>
        <Unknown
          href="https://mc.commercetools.com/a-cool-project/products/2c7f2de9-d30a-42cd-bbca-5783381efb9c/general"
          target="_blank"
        >
          Open in Merchant Center
        </Unknown>
      </Unknown>
    </React.Fragment>
  }
  loading={false}
  thumbnailElement={
    <img
      alt=""
      height="70"
      src="https://s3-eu-west-1.amazonaws.com/commercetools-maximilian/products/073315_1_large.jpg"
      width="70"
    />
  }
  title="Shirt Red Valentino red (ABCD1234)"
/>
`;

exports[`ProductCardRenderer renders with data and disabled 1`] = `
<div
  cardDragHandleComponent={null}
  className="entry-card"
  contentType="T-shirts"
  dropdownListElements={
    <React.Fragment>
      <Unknown>
        <Unknown
          isDisabled={false}
          onClick={[Function]}
        >
          Remove
        </Unknown>
        <Unknown
          href="https://mc.commercetools.com/a-cool-project/products/2c7f2de9-d30a-42cd-bbca-5783381efb9c/general"
          target="_blank"
        >
          Open in Merchant Center
        </Unknown>
      </Unknown>
    </React.Fragment>
  }
  loading={false}
  thumbnailElement={
    <img
      alt=""
      height="70"
      src="https://s3-eu-west-1.amazonaws.com/commercetools-maximilian/products/073315_1_large.jpg"
      width="70"
    />
  }
  title="Shirt Red Valentino red (ABCD1234)"
/>
`;

exports[`ProductCardRenderer renders with data and sortable 1`] = `
<div
  cardDragHandleComponent={<DragHandle />}
  className="entry-card"
  contentType="T-shirts"
  dropdownListElements={
    <React.Fragment>
      <Unknown>
        <Unknown
          isDisabled={false}
          onClick={[Function]}
        >
          Remove
        </Unknown>
        <Unknown
          href="https://mc.commercetools.com/a-cool-project/products/2c7f2de9-d30a-42cd-bbca-5783381efb9c/general"
          target="_blank"
        >
          Open in Merchant Center
        </Unknown>
      </Unknown>
    </React.Fragment>
  }
  loading={false}
  thumbnailElement={
    <img
      alt=""
      height="70"
      src="https://s3-eu-west-1.amazonaws.com/commercetools-maximilian/products/073315_1_large.jpg"
      width="70"
    />
  }
  title="Shirt Red Valentino red (ABCD1234)"
/>
`;

exports[`ProductCardRenderer renders with data and sortable and disabled 1`] = `
<div
  cardDragHandleComponent={<DragHandle />}
  className="entry-card"
  contentType="T-shirts"
  dropdownListElements={
    <React.Fragment>
      <Unknown>
        <Unknown
          isDisabled={false}
          onClick={[Function]}
        >
          Remove
        </Unknown>
        <Unknown
          href="https://mc.commercetools.com/a-cool-project/products/2c7f2de9-d30a-42cd-bbca-5783381efb9c/general"
          target="_blank"
        >
          Open in Merchant Center
        </Unknown>
      </Unknown>
    </React.Fragment>
  }
  loading={false}
  thumbnailElement={
    <img
      alt=""
      height="70"
      src="https://s3-eu-west-1.amazonaws.com/commercetools-maximilian/products/073315_1_large.jpg"
      width="70"
    />
  }
  title="Shirt Red Valentino red (ABCD1234)"
/>
`;

exports[`SortableProductCard renders correctly 1`] = `
<div
  className="query"
  query={
    Object {
      "definitions": Array [
        Object {
          "directives": Array [],
          "kind": "OperationDefinition",
          "name": Object {
            "kind": "Name",
            "value": "productsList",
          },
          "operation": "query",
          "selectionSet": Object {
            "kind": "SelectionSet",
            "selections": Array [
              Object {
                "arguments": Array [
                  Object {
                    "kind": "Argument",
                    "name": Object {
                      "kind": "Name",
                      "value": "skus",
                    },
                    "value": Object {
                      "kind": "Variable",
                      "name": Object {
                        "kind": "Name",
                        "value": "skus",
                      },
                    },
                  },
                ],
                "directives": Array [],
                "kind": "Field",
                "name": Object {
                  "kind": "Name",
                  "value": "products",
                },
                "selectionSet": Object {
                  "kind": "SelectionSet",
                  "selections": Array [
                    Object {
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "results",
                      },
                      "selectionSet": Object {
                        "kind": "SelectionSet",
                        "selections": Array [
                          Object {
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "id",
                            },
                          },
                          Object {
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "masterData",
                            },
                            "selectionSet": Object {
                              "kind": "SelectionSet",
                              "selections": Array [
                                Object {
                                  "arguments": Array [],
                                  "directives": Array [],
                                  "kind": "Field",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "current",
                                  },
                                  "selectionSet": Object {
                                    "kind": "SelectionSet",
                                    "selections": Array [
                                      Object {
                                        "arguments": Array [],
                                        "directives": Array [],
                                        "kind": "Field",
                                        "name": Object {
                                          "kind": "Name",
                                          "value": "categories",
                                        },
                                        "selectionSet": Object {
                                          "kind": "SelectionSet",
                                          "selections": Array [
                                            Object {
                                              "arguments": Array [],
                                              "directives": Array [],
                                              "kind": "Field",
                                              "name": Object {
                                                "kind": "Name",
                                                "value": "nameAllLocales",
                                              },
                                              "selectionSet": Object {
                                                "kind": "SelectionSet",
                                                "selections": Array [
                                                  Object {
                                                    "arguments": Array [],
                                                    "directives": Array [],
                                                    "kind": "Field",
                                                    "name": Object {
                                                      "kind": "Name",
                                                      "value": "locale",
                                                    },
                                                  },
                                                  Object {
                                                    "arguments": Array [],
                                                    "directives": Array [],
                                                    "kind": "Field",
                                                    "name": Object {
                                                      "kind": "Name",
                                                      "value": "value",
                                                    },
                                                  },
                                                ],
                                              },
                                            },
                                          ],
                                        },
                                      },
                                      Object {
                                        "arguments": Array [],
                                        "directives": Array [],
                                        "kind": "Field",
                                        "name": Object {
                                          "kind": "Name",
                                          "value": "nameAllLocales",
                                        },
                                        "selectionSet": Object {
                                          "kind": "SelectionSet",
                                          "selections": Array [
                                            Object {
                                              "arguments": Array [],
                                              "directives": Array [],
                                              "kind": "Field",
                                              "name": Object {
                                                "kind": "Name",
                                                "value": "locale",
                                              },
                                            },
                                            Object {
                                              "arguments": Array [],
                                              "directives": Array [],
                                              "kind": "Field",
                                              "name": Object {
                                                "kind": "Name",
                                                "value": "value",
                                              },
                                            },
                                          ],
                                        },
                                      },
                                      Object {
                                        "arguments": Array [],
                                        "directives": Array [],
                                        "kind": "Field",
                                        "name": Object {
                                          "kind": "Name",
                                          "value": "masterVariant",
                                        },
                                        "selectionSet": Object {
                                          "kind": "SelectionSet",
                                          "selections": Array [
                                            Object {
                                              "arguments": Array [],
                                              "directives": Array [],
                                              "kind": "Field",
                                              "name": Object {
                                                "kind": "Name",
                                                "value": "sku",
                                              },
                                            },
                                            Object {
                                              "arguments": Array [],
                                              "directives": Array [],
                                              "kind": "Field",
                                              "name": Object {
                                                "kind": "Name",
                                                "value": "images",
                                              },
                                              "selectionSet": Object {
                                                "kind": "SelectionSet",
                                                "selections": Array [
                                                  Object {
                                                    "arguments": Array [],
                                                    "directives": Array [],
                                                    "kind": "Field",
                                                    "name": Object {
                                                      "kind": "Name",
                                                      "value": "url",
                                                    },
                                                  },
                                                ],
                                              },
                                            },
                                          ],
                                        },
                                      },
                                    ],
                                  },
                                },
                              ],
                            },
                          },
                        ],
                      },
                    },
                  ],
                },
              },
            ],
          },
          "variableDefinitions": Array [
            Object {
              "directives": Array [],
              "kind": "VariableDefinition",
              "type": Object {
                "kind": "ListType",
                "type": Object {
                  "kind": "NonNullType",
                  "type": Object {
                    "kind": "NamedType",
                    "name": Object {
                      "kind": "Name",
                      "value": "String",
                    },
                  },
                },
              },
              "variable": Object {
                "kind": "Variable",
                "name": Object {
                  "kind": "Name",
                  "value": "skus",
                },
              },
            },
          ],
        },
      ],
      "kind": "Document",
      "loc": Object {
        "end": 457,
        "source": Object {
          "body": "query productsList($skus: [String!]) {
  products(skus: $skus) {
    results {
      id
      masterData {
        current {
          categories {
            nameAllLocales {
              locale
              value
            }
          }
          nameAllLocales {
            locale
            value
          }
          masterVariant {
            sku
            images {
              url
            }
          }
        }
      }
    }
  }
}
",
          "locationOffset": Object {
            "column": 1,
            "line": 1,
          },
          "name": "GraphQL request",
        },
        "start": 0,
      },
    }
  }
  variables={
    Object {
      "skus": Array [
        "ABCD1234",
      ],
    }
  }
>
  [object Function]
</div>
`;
